plugins/plugin-chart-table/src/DataTable/hooks/useSticky.tsx

```diff
@@ -0,0 +1,264 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, {
+  useRef,
+  useLayoutEffect,
+  useCallback,
+  ReactNode,
+  ReactElement,
+  ComponentPropsWithRef,
+  CSSProperties,
+  UIEventHandler,
+} from 'react';
+import { TableInstance, Hooks, ensurePluginOrder } from 'react-table';
+
+type ReactElementWithChildren<
+  T extends keyof JSX.IntrinsicElements,
+  C extends ReactNode = ReactNode
+> = ReactElement<ComponentPropsWithRef<T> & { children: C }, T>;
+
+type th = ReactElementWithChildren<'th'>;
+type td = ReactElementWithChildren<'td'>;
+type trWithTh = ReactElementWithChildren<'tr', th>;
+type trWithTd = ReactElementWithChildren<'tr', td>;
+type thead = ReactElementWithChildren<'thead', trWithTh>;
+type tbody = ReactElementWithChildren<'tbody', trWithTd>;
+type col = ReactElementWithChildren<'col', null>;
+type colgroup = ReactElementWithChildren<'colgroup', col>;
+type table = ReactElementWithChildren<'table', (thead | tbody | colgroup)[]>;
+
+export interface StickyElementSize {
+  width?: number; // full table width
+  height?: number; // full table height
+  bodyWidth?: number; // scrollable area width
+  bodyHeight?: number; // scrollable area height
+  columnWidths?: number[];
+}
+
+export interface UseStickyTableOptions {
+  getTableSize?: () => Partial<StickyElementSize> | undefined;
+}
+
+export interface UseStickyInstanceProps {
+  // HOC for manipulating DOMs in <table> to make the header sticky
+  StickyWrap: React.FunctionComponent<{ children: table }>;
+  // update or recompute the sticky table size
+  setStickyElementSize: (size?: StickyElementSize) => void;
+}
+
+export type UseStickyState = {
+  sticky: StickyElementSize;
+};
+
+export enum ReducerActions {
```

Comment:
generally we don't use `enum`s ever, not sure if these have crept into `superset-ui` or not but I would use a standard `const` `object` instead.

Reply:
Have both of you resolved this?

Reply:
I'd still be in favor of `enum`s if possible. Even Airbnb's internal typescript style guide didn't explicitly ban them.